File: Java/src/Solution401_500/Solution495.java
package Solution401_500;

/**
 * @author windows
 */
public class Solution495 {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int count = 0;
        for (int i = 0; i < timeSeries.length - 1; i++) {
            if (timeSeries[i] - timeSeries[i + 1] < duration){
                count += timeSeries[i + 1] - timeSeries[i];
            }else{
                count += duration;
            }
        }
        return count + duration;
    }


    public static void main(String[] args) {
        Solution495 solution495 = new Solution495();
    }
}

File: C++/src/Solution401_500/Solution495.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution495.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution495::findPoisonedDuration(vector<int> &timeSeries, int duration) {
    int total = 0;
    for (int i = 1; i < timeSeries.size(); ++i) {
        if (timeSeries[i] - timeSeries[i - 1] < duration){
            total += timeSeries[i] - timeSeries[i - 1];
        }else{
            total += duration;
        }
    }
    return total + duration;
}
